### PR TITLE
Ele 3893 alert rules operators

### PR DESCRIPTION
**Pull request by MikaKerman**
null

### CODE REVIEW

File: elementary/monitor/api/alerts/alert_filters.py [hunk 1]
```diff
@@ -16,6 +16,80 @@
...
+        result.append(splitter.join(parts[i:]))
+
+    return result
+
+
+def _get_alert_node_name(alert: PendingAlertSchema) -> Optional[str]:
+    alert_node_name = None
+    alert_type = AlertTypes(alert.type)
+    if alert_type is AlertTypes.TEST:
+        alert_node_name = alert.data.test_name  # type: ignore[union-attr]
```
Comment (ofek1weiss) [inline elementary/monitor/api/alerts/alert_filters.py:35]: i think you can remove this type ignore by adding type hints in line 32: `alert_node_name: Optional[str] = None`

File: elementary/monitor/data_monitoring/schema.py [hunk 2]
```diff
@@ -14,46 +14,88 @@ class InvalidSelectorError(Exception):
...
+        if self.type in ANY_OPERATORS:
+            return any(self.apply_filter_on_value(value) for value in values)
+        elif self.type in ALL_OPERATORS:
+            return all(self.apply_filter_on_value(value) for value in values)
+        raise ValueError(f"Unsupported filter type: {self.type}")
+
 
-class StatusFilterSchema(FilterSchema):
+class StatusFilterSchema(FilterSchema[Status]):
     values: List[Status]
```
Comment (ofek1weiss) [inline elementary/monitor/data_monitoring/schema.py:88]: you can remove the re-definition of `values` here

File: elementary/monitor/data_monitoring/schema.py [hunk 3]
```diff
@@ -14,46 +14,88 @@ class InvalidSelectorError(Exception):
...
 
 
-class ResourceTypeFilterSchema(FilterSchema):
+class ResourceTypeFilterSchema(FilterSchema[ResourceType]):
     values: List[ResourceType]
```
Comment (ofek1weiss) [inline elementary/monitor/data_monitoring/schema.py:92]: here too

File: elementary/monitor/api/alerts/alert_filters.py [hunk 4]
```diff
@@ -16,6 +16,80 @@
...
+    resource_type = ResourceType(alert.data.resource_type)
+
+    alert_node_name = _get_alert_node_name(alert)
+    node_names = (
+        [alert_node_name, *get_string_ends(alert_node_name, ".")]
+        if alert_node_name
+        else []
+    )
+
+    return (
```
Comment (ofek1weiss) [inline elementary/monitor/api/alerts/alert_filters.py:62]: move this to separate function

Comment (MikaKerman) [reply elementary/monitor/api/alerts/alert_filters.py]: Done.

File: elementary/monitor/data_monitoring/schema.py (hunk 2 above)
Comment (MikaKerman) [reply elementary/monitor/data_monitoring/schema.py:88]: I thought so too, but apparently not.
https://github.com/pydantic/pydantic/issues/4171

File: elementary/monitor/data_monitoring/schema.py (hunk 3 above)
Comment (MikaKerman) [reply elementary/monitor/data_monitoring/schema.py:92]: Commented above.

File: elementary/monitor/api/alerts/alert_filters.py (hunk 1 above)
Comment (MikaKerman) [reply elementary/monitor/api/alerts/alert_filters.py:31]: Apparently not :( 
```
def _get_alert_node_name(alert: PendingAlertSchema) -> Optional[str]:
    alert_node_name: Optional[str] = None
    alert_type = AlertTypes(alert.type)
    if alert_type is AlertTypes.TEST:
        alert_node_name = alert.data.test_name
```

```
elementary/monitor/api/alerts/alert_filters.py:31: error: Item "ModelAlertDataSchema" of "Union[TestAlertDataSchema, ModelAlertDataSchema, SourceFreshnessAlertDataSchema]" has no attribute "test_name"  [union-attr]
elementary/monitor/api/alerts/alert_filters.py:31: error: Item "SourceFreshnessAlertDataSchema" of "Union[TestAlertDataSchema, ModelAlertDataSchema, SourceFreshnessAlertDataSchema]" has no attribute "test_name"  [union-attr]
```

It's from the [earlier version](https://github.com/elementary-data/elementary/pull/1768/files#diff-6f78902926e4d87f966d446498d442d9c2f33492171548b1a8fc72b25e49727bL153).

**Review by ofek1weiss — APPROVED** [review]

**Pull request merged**